coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -71,17 +78,73 @@ type Server struct {
 	TimeNowFn func() time.Time
 }
 
+func NewServer(
+	accessURL *url.URL,
+	id uuid.UUID,
+	logger slog.Logger,
+	provisioners []database.ProvisionerType,
+	tags json.RawMessage,
+	db database.Store,
+	ps pubsub.Pubsub,
+	tel telemetry.Reporter,
+	tracer trace.Tracer,
+	quotaCommitter *atomic.Pointer[proto.QuotaCommitter],
+	auditor *atomic.Pointer[audit.Auditor],
+	templateScheduleStore *atomic.Pointer[schedule.TemplateScheduleStore],
+	userQuietHoursScheduleStore *atomic.Pointer[schedule.UserQuietHoursScheduleStore],
+	deploymentValues *codersdk.DeploymentValues,
+	acquireJobDebounce time.Duration,
+	options Options,
+) proto.DRPCProvisionerDaemonServer {
+	// Panic early if pointers are nil
+	if quotaCommitter == nil {
+		panic("quotaCommitter is nil")
+	}
+	if auditor == nil {
+		panic("auditor is nil")
+	}
+	if templateScheduleStore == nil {
+		panic("templateScheduleStore is nil")
+	}
+	if userQuietHoursScheduleStore == nil {
+		panic("userQuietHoursScheduleStore is nil")
+	}
+	if deploymentValues == nil {
+		panic("deploymentValues is nil")
+	}
```

Comment:
I would rather we return errors here so this can be handled more gracefully at the calling site.

Let's say we introduce a regression where, in a certain configuration, some of these are passed as nil. So our tests don't catch it, but our customer will. By the time these panic, `coderd` could've started some work that will be canceled abruptly (hypothetically, probably not the way todays server starts up, but in the future).